src/remark-filter-nodes.js

```diff
@@ -0,0 +1,48 @@
+const visit = require('unist-util-visit')
+
+const splice = [].splice
+
+module.exports = remarkFilterNodes
+
+function remarkFilterNodes(options) {
+  if (options.allowedTypes && options.disallowedTypes) {
+    throw new Error('Only one of `allowedTypes` and `disallowedTypes` should be defined')
+  }
+
+  return options.allowedTypes || options.disallowedTypes || options.allowNode
+    ? transform
+    : undefined
+
+  function transform(tree) {
+    visit(tree, onnode)
+  }
+
+  function onnode(node, index, parent) {
+    let remove
+
+    if (parent) {
+      if (options.allowedTypes) {
+        remove = !options.allowedTypes.includes(node.type)
+      } else if (options.disallowedTypes) {
+        remove = options.disallowedTypes.includes(node.type)
+      }
+
+      if (!remove && options.allowNode) {
+        remove = !options.allowNode(node, index, parent)
+      }
+    }
+
+    if (remove) {
+      let parameters = [index, 1]
+
+      if (options.unwrapDisallowed && node.children) {
+        parameters = parameters.concat(node.children)
+      }
+
+      splice.apply(parent.children, parameters)
```

Comment:
would this be the same as

```suggestion
      parent.children = [...parent.children, ...parameters]
```

Reply:
Yeah, although I’m assuming splice is faster though because it mutates (see micromark things a while ago).
Anyway, this code was already here, I copied it in from a now-removed file. I’m not sure what the baseline for react-markdown and es version is, because there’s also `xtend` instead of `Object.assign`.
Maybe something for the ESM move?

Reply:
> Maybe something for the ESM move?

fine by me